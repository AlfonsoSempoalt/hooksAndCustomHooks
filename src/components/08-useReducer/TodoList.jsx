import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todos = [], handleToggle, handleOnDelete }) => {
  return (
    <>
      <ul className='list-group list-group-flush'>
        {todos.map((todo, i) => (
          <TodoListItem
            todo={todo}
            key={todo.id}
            handleOnDelete={handleOnDelete}
            handleToggle={handleToggle}
            index={i}
          />
        ))}
      </ul>
    </>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleOnDelete: PropTypes.func.isRequired,
};

export default TodoList;
