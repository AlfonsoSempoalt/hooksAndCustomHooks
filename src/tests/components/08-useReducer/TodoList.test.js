import React from 'react';
import { shallow } from 'enzyme';
import { demoTodos } from '../fixtures/demoTodos';
import TodoList from '../../../components/08-useReducer/TodoList';

describe('Name of the group', () => {
    const handleToggle = jest.fn();
    const handleOnDelete = jest.fn();

    const wrapper = shallow(
        <TodoList
            todos={demoTodos}
            handleToggle={handleToggle}
            handleOnDelete={handleOnDelete}
        />
    );

    test('should match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have two ListItems', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(
            wrapper.find('TodoListItem').at(0).prop('handleOnDelete')
        ).toEqual(expect.any(Function));
    });
});
