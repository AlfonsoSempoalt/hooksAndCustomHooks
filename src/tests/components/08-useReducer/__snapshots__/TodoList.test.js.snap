// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should match with the snapshot 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleOnDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "desc": "aprender react",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleOnDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "desc": "hola",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
